tools/crm_resource.c

```diff
@@ -1625,23 +1625,25 @@ main(int argc, char **argv)
 
         CRM_ASSERT(len > 0);
 
-        strv = calloc(len, sizeof(char *));
-        strv[0] = strdup("non-option ARGV-elements:");
+        /* Add 1 for the strv[0] string below, and add another 1 for the NULL
+         * at the end of a string vector.
+         */
+        strv = calloc(len+2, sizeof(char *));
+        strv[0] = strdup("non-option ARGV-elements: ");
 
         for (gchar **s = options.remainder; *s; s++) {
             strv[i] = crm_strdup_printf("[%d of %d] %s\n", i, len, *s);
             i++;
         }
 
+        strv[i] = NULL;
+
         exit_code = CRM_EX_USAGE;
         msg = g_strjoinv("", strv);
         g_set_error(&error, PCMK__EXITC_ERROR, exit_code, "%s", msg);
-        g_free(msg);
 
-        for(i = 0; i < len; i++) {
-            free(strv[i]);
-        }
-        free(strv);
+        g_free(msg);
+        g_strfreev(strv);
```

Comment:
Actually this needs to stay the same since it was allocated by libc rather than glib

Reply:
Fixed.